src/lib/mina_base/party.ml

```diff
@@ -165,6 +185,54 @@ module Update = struct
     end
   end]
 
+  let gen : t Quickcheck.Generator.t =
+    let open Quickcheck.Let_syntax in
+    let%bind app_state =
```

Comment:
We need to know here that the permission matches the authentication kind for success, for guaranteed success cases we may need to do something involving the ledger to generate these.

Reply:
@mrmr1993 Can you provide more detail?

Reply:
In the account there is a `permissions` record that sets out the kind of permission for each potential update. If we attempt to use a signature to issue an update that has `Proof` auth required, that predicate will fail. (We can probably punt on this for now and do it later, what you have here is already great for testing.)